test/yargs-parser.js

```diff
@@ -718,6 +719,7 @@ describe('yargs-parser', function () {
 
     it('should load nested options from config object', function () {
       var argv = parser(['--nested.foo', 'bar'], {
+        configuration: { 'config-dot-notation': true },
         configObjects: [{
           a: 'a',
           nested: {
```

Comment:
It would be good to add two tests, demonstrating "foo.bar" with `config-dot-notation` enabled and disabled.

Reply:
Yeah I was thinking of adding a test but I saw "does not append nested-object keys from config to top-level key" seems to be testing exactly that.

Except, now that I look at it again, the only thing I would change is line 2100 change `'dot-notation': false` to `'config-dot-notation': false` to make it explicit.

Or would you rather like me to add another test case for this?